File: typescript/src/services/ofertas/ofertas.contract.ts
import type { OfertasRequest, OfertasResponse } from "../../api/ofertas/ofertas.dto.js"

/** Contrato do motor de crédito — implementado em `ofertas.service.ts`. */
export interface OfertasContract {
  gerar(request: OfertasRequest): Promise<OfertasResponse> | OfertasResponse
}

File: java/src/main/java/br/com/atlas/averbacao/services/ofertas/OfertasContract.java
package br.com.atlas.averbacao.services.ofertas;

import br.com.atlas.averbacao.api.ofertas.OfertasRequest;
import br.com.atlas.averbacao.api.ofertas.OfertasResponse;

public interface OfertasContract {
    OfertasResponse gerar(OfertasRequest request);
}
